### PR TITLE
CORE-868: switch to keycloak

### PR DESCRIPTION
**Pull request by slr71**
This PR switches Keycloak from CAS to Keycloak. The features that we needed are all readily available in the official `keycloak-connect` authentication module, so `passport` and its auth strategies have been removed as dependencies.

### CODE REVIEW

**Comment by slr71** [comment]
I just did a rebase to ensure that these changes can be merged without any issues.

File: src/server/api/router.js [hunk 1]
```diff
@@ -49,21 +49,12 @@ export default function apiRouter() {
...
-        if (req.user) {
-            res.json({
-                id: req.user.profile.id,
-                attributes: req.user.profile.attributes,
-            });
-        } else {
-            res.json(null);
-        }
-    });
+    api.get("/profile", (req, res) => res.json(auth.getUserProfile(req)));
```
Comment (psarando) [inline src/server/api/router.js]: 👍

File: src/server/index.js [hunk 2]
```diff
@@ -55,46 +61,30 @@ app.prepare()
...
+        server.get("/login", keycloakClient.protect());
 
         logger.info("adding the /login/* handler");
-        server.get("/login/*", ensureLoggedIn("/login"), (req, res) => {
+        server.get("/login/*", keycloakClient.protect(), (req, res) => {
             res.redirect(req.url.replace(/^\/login/, ""));
         });
 
-        logger.info("adding the /logout handler");
-        server.get("/logout", (req, res) => {
```
Comment (psarando) [inline src/server/index.js]: So keycloak doesn't support logging out?

File: src/server/logging.js [hunk 3]
```diff
@@ -15,23 +16,24 @@ const logger = createLogger({
     transports: [new transports.Console()],
 });
 
+const getLoggableUserID = (req) => getUserID(req) || "logged-out-user";
```
Comment (psarando) [inline src/server/logging.js]: 👍

File: src/server/index.js (hunk 2 above)
Comment (slr71) [reply src/server/index.js]: It does; the logout endpoint is built into the middleware, so we don't have to define it explicitly anymore.

**Comment by slr71** [comment]
I'm running one more test against terrain in QA just to make sure it has a good chance of working. 🤞 If it works, I'll merge this PR.

**Pull request merged**